src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -32,6 +32,9 @@ class LogicalPlan:
     def __init__(self) -> None:
         self.children = []
         self._cumulative_node_complexity: Optional[Dict[PlanNodeCategory, int]] = None
+        # This flag is used to determine if the current node is a valid candidate for
+        # replacement in plan tree during optimization stage using replace_child method.
```

Comment:
should we add a comment here for deepcopy?

Reply:
added